revscoring/datasources/meta/vectorizers.py

```diff
@@ -6,42 +6,41 @@
 """
 import os.path
 
-import fasttext
-
 from gensim.models.keyedvectors import KeyedVectors
 
 from ..datasource import Datasource
 
 ASSET_SEARCH_DIRS = ["word2vec/", "~/.word2vec/", "/var/share/word2vec/"]
-VECTOR_DIMENSIONS = 300
 
 
 class word2vec(Datasource):
     """
-    Generates vectors for a list of items generated by another
-    datasource.
+    Generates vectors using "word2vec format" for a list of items generated
+    by another datasource.
 
     :Parameters:
         words_datasource : :class:`revscoring.Datasource`
             A datasource that returns a list of words.
         vectorize_words : `function`
             a function that takes a list of words and converts it to a list
             of vectors of those words
+        keyed_vectors : `dict`
+            A list of word vectors in "word2vec format"
         name : `str`
             A name for the `revscoring.FeatureVector`
     """  # noqa
 
-    def __init__(self, words_datasource, vectorize_words, name=None):
+    def __init__(self, words_datasource, vectorize_words, keyed_vectors, name=None): 
         name = self._format_name(name, [words_datasource, vectorize_words])
 
         super().__init__(name, vectorize_words, depends_on=[words_datasource])
 
     @staticmethod
     def vectorize_words(keyed_vectors, words):
```

Comment:
Let's change this to a regular instance method. 

```
def process(self, words):
    if not words:
        return [[0] * self.keyed_vectors.vector_size]
    ...
```

This will mean that we change the usage to look like this:
```
keyed_vectors = {...}
w2v = word2vec(words_datasource, keyed_vectors)
solve(w2v, cache={words_datasource: ["I", "am", "some", "words"])
```